build-system/tasks/extension-generator/extension-doc.template.md

```diff
@@ -0,0 +1,220 @@
+---
+\$category@: presentation
+formats:
+  - websites
+  - email
+teaser:
+  text: Fill this in with teaser text to improve SEO. Use the component description.
+---
+
+<!--
+  All documentation starts with frontmatter. Front matter organizes documentation on amp.dev
+  and improves SEO.
+  * Include the relevant category(ies): ads-analytics, dynamic-content, layout, media, presentation, social, personalization
+  * List applicable format(s): websites, ads, stories, email
+  * Do not include markdown formatting in the frontmatter - plain text and punctionation only!
+  * Remove this comment!
+-->
+
+<!--
+Copyright ${year} The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# \${name}
+
+<!--
+  If the component is relevant for more than one format and operates differently between these
+  formats, include and filter multiple content blocks and code samples.
+-->
+
+## Usage
+
+One to three paragraphs explaining the component usage. List important functionality. Explain why developers care about it.
+
+[filter formats=“websites”]
+
+Below is an example for websites.
+
+[example preview="inline" playground="true" imports="${name}"]
+
+```html
+<${name} required-attribute>
+  I am a hello world inline executable code sample for websites!
+</${name}>
+```
+
+[/example][/filter]
+
+<!--
+  * [Read more about filtering sections](https://amp.dev/documentation/guides-and-tutorials/contribute/contribute-documentation/formatting/?format=websites#filtering-sections)
+  * [Read more about executable code samples](https://amp.dev/documentation/guides-and-tutorials/contribute/contribute-documentation/formatting/?format=websites#preview-code-samples)
+ -->
+
+[filter formats=“ads”]
+
+Below is an example for ads.
+
+[example preview=“inline” playground=“true” imports="${name}"]
+
+```html
+<${name} required-attribute>
+  I am a hello world inline executable code sample for ads!
+</${name}>
+```
+
+[/example][/filter]
+
+### Standalone use outside valid AMP documents (optional)
```

Comment:
@rcebulko can correct me if I'm wrong, but won't this file be duplicated for non-bento components, and the other file in this PR (`build-system/tasks/extension-generator/bento/amp-__component_name_hyphenated__/amp-__component_name_hyphenated__.md`) for Bento components? 

In other words, should "Standalone use outside valid AMP documents" be moved to `build-system/tasks/extension-generator/bento/amp-__component_name_hyphenated__/amp-__component_name_hyphenated__.md`?

Reply:
Both legacy (AMP) and Bento generator use the `amp-__component_name_hyphenated__.md` file, so yes anything added here should be move to that file, and this file can be removed entirely. This file currently isn't even read by the generator.